test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -74,22 +76,96 @@ class Assertion[-A] private (render: String, val run: (=> A) => AssertResult[Eit
    * Provides a meaningful string rendering of the assertion.
    */
   override final def toString: String =
-    render
+    render.toString
 }
 
 object Assertion {
 
+  /**
+   * `Render` captures both the name of an assertion as well as the parameters
+   * to the assertion combinator for pretty-printing.
+   */
+  sealed trait Render {
+    override final def toString: String = this match {
+      case Render.Function(name, paramLists) =>
+        name + paramLists.map(_.mkString("(", ", ", ")")).mkString
+      case Render.Infix(left, op, right) =>
+        "(" + left + " " + op + " " + right + ")"
```

Comment:
This is good. Eventually we could embed precedence here to make rendering a bit nicer.